File: src/lib/libJKFactory/ga-5-3b/armci/examples/benchmarks/cg/armci_sharedmemory/generateinput.f
        program matrix_gen
        integer*4 i, k
        double precision amat(8,8), xvec(8), bvec(8)
        integer *4 irow(8), icol(8*8)
        integer*4 timeArray(3), ei,sf;
        open (unit = 8, file = "matrix.bin")
        ei = 8
        sf = 64
        do i = 1, 8
          do k = 1, 8
            do n=1,100000
              call itime(timeArray)
              amat(i,k) = amat(i,k) + 1212.12*
     &              rand(timeArray(1)/(timeArray(2)+timeArray(3)))
            end do
            call itime(timeArray)
            amat(i,k) = amat(i,k)+rand ( timeArray(1)+timeArray(2)+
     &                      timeArray(3))
          end do
          call itime(timeArray)
          xvec(i) = rand ( timeArray(1)+timeArray(2)+timeArray(3) )
        end do
        do i = 1,8
          bvec(i)=0D0
          irow(i)=(i-1)*8+1
          do k =1, 8
            bvec(i) = bvec(i)+xvec(k)*amat(i,k)
            icol((i-1)*8+k)=i
          end do
        end do
        write(8,*) ei
        write(8,*) sf
        write(8,*) amat 
        write(8,*) irow
        write(8,*) icol
        write(8,*) xvec
        close(8)
        write(6,*) 'amat =', amat 
        write(6,*) 'xvec =', xvec
        write(6,*) 'bvec=', bvec
        write(6,*) 'irow=', irow
        write(6,*) 'icol=', icol
        stop
        end

File: src/lib/libJKFactory/ga-5-3b/armci/examples/benchmarks/cg/armci_sharedmemory/generateinput.c
#if HAVE_CONFIG_H
#   include "config.h"
#endif

#if HAVE_STDLIB_H
#   include <stdlib.h>
#endif
#if HAVE_SYS_TYPES_H
#   include <sys/types.h>
#endif
#if HAVE_SYS_STAT_H
#   include <sys/stat.h>
#endif
#if HAVE_FCNTL_H
#   include <fcntl.h>
#endif
#if HAVE_STDIO_H
#   include <stdio.h>
#endif

void printsparse_(int *irow, int *icol, double *a,  double *b, int n,int nz)
{
int fd;
int i,max,j;

    fd = open("matrix.bin",O_CREAT|O_WRONLY,0777);
    if(fd<=0) { printf("open failed"); return;}

    write (fd, &n, sizeof(int));
    write (fd, &nz, sizeof(int));
    write (fd, a, sizeof(double)*(nz));
    write (fd, irow, sizeof(int)*(n));
    write (fd, icol, sizeof(int)*(nz));
    write (fd, b, sizeof(double)*(n));

    close(fd);

    printf("dumped sparse matrix: dim = %d %d nonzeros\n", n, nz);

}

int main(int argc, char **argv)
{
int NZROW=8,NZSQ=64;
int i,k,irow[NZROW+1],icol[NZSQ+4],n=NZROW,nz=NZSQ;
double    a[NZSQ],  b[NZROW], xvec[NZROW];
    for(i=0;i<NZROW;i++){
      for(k=0;k<NZROW;k++)
        a[i*NZROW+k] =  k+1;
      xvec[i] = i+3.5;
    }
    for(i=0;i<NZROW;i++){
      irow[i]=i*NZROW+1;
      b[i]=0;
      for(k=0;k<NZROW;k++){
        b[i] = b[i]+xvec[k]*a[i*NZROW+k];
        icol[i*NZROW+k]=i+1;
      }
      printf("\nb[%d]=%f",i,b[i]);
    }
    printsparse_(irow,icol,a,b,n,nz);
    return(1);
}
